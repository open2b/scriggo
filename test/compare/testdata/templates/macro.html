{# render #}

{% macro M1 %}M1{% return %}{% end macro %}

{% macro M2(int, int) %}M2{%% return %%}M2{% end macro %}

{% macro M3(a string) %}called M3 with argument '{{ a }}'{% end macro %}

{% macro M4() html %}<b>M4</b>{% return %}<b>M4</b>{% end macro %}

{{ M1() }}
{{ M2(20, 20) }}
{{ M3("a string!") }}
{{ M4() }}
{% s := M4() %}{{ M4() }}
