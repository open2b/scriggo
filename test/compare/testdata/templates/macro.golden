M1
M2
called M3 with argument 'a string!'
<b>M4</b>
<b>M4</b>